meson-mode.el

```diff
@@ -914,21 +914,25 @@ Return either `line-beginning-position' of the matching line or nil."
   (and (re-search-forward (meson--make-lookup-regexp identifier) nil t)
        (line-beginning-position)))
 
-(defun meson-lookup-doc (what)
-  "Open Meson reference manual and find the function or object named WHAT."
-  (when-let (refman (seq-find 'file-exists-p
-			      (mapcar (lambda (file) (expand-file-name file meson-markdown-docs-dir))
-				      '("Reference-manual.md"
-					"Reference-manual.md.gz"))))
-    (find-file-read-only refman)
+(defun meson-lookup-doc (identifier)
+  "Open Meson reference manual and find the function or object named IDENTIFIER.
+Return the buffer containing the reference manual or nil."
+  (when-let* ((refman (meson--find-reference-manual))
+	      (buf (find-file-noselect refman))
+	      (pos (with-current-buffer buf
+		     (meson--search-in-reference-manual identifier))))
```

Comment:
It would be nice to have a `if-let*` here (maybe more than one) with else branch(es) containing error `(message)` like "Meson reference manual not found" and "%s not found in Meson reference manual".

Reply:
I’ve changed the `when-let*` to a `let*` signaling `user-error` in 5f5c1292186ed226ed5ef27e55c2622c80022102.